rstan/rstan/R/misc.R

```diff
@@ -1604,6 +1604,10 @@ parse_data <- function(cppcode) {
                   cppcode[private:public])
   # get them from the calling environment
   objects <- objects[nzchar(trimws(objects))]
+  # Remove model internal name underscores in case of Eigen::Maps
+  objects = gsub("([0-9A-Za-z_]+)__", "\\1", objects)
+  # Remove any bad regex matches that found the end of an Eigen::Map.
+  objects = objects[suppressWarnings(is.na(as.numeric(objects)))]
```

Comment:
```suggestion
  # Remove model internal name underscores in case of Eigen::Maps
  objects <- gsub("([0-9A-Za-z_]+)__", "\\1", objects)
  # Remove any bad regex matches that found the end of an Eigen::Map.
  objects <- gsub("^[[:digit:]]+", "\\1", objects)
  # Remove empty characters and trim whitespaces
  objects <- objects[nzchar(trimws(objects))]
  
  # Get them from the calling environment
```

- Use right-to-left assignment
- Remove all objects that starts with a number
- Remove empty characters and trim whitespaces at the end